src/components/dock/dock.types.ts

```diff
@@ -1,75 +1,88 @@
 export interface DockItem {
-    expanded: boolean;
-    useMobileLayout: boolean;
-}
+    /**
+     * A non-changing value to uniquely identify each item.
+     */
+    id: string;
 
-export interface DockItemConfig<T = any> {
     /**
      * Text to display for the item.
      */
     label: string;
 
     /**
-     * xxxx
+     * Name of the icon to use.
+     */
+    icon: string;
+
+    /**
+     * Additional helper text for the button.
+     * Example usage can be a keyboard shortcut to activate the button.
      */
     helperLabel?: string;
 
     /**
-     * True if the list item should be selected.
+     * Whether the button should indicate it is selected.
+     * The button should be selected when … (@Kiarokh: add description here please!)
      */
     selected?: boolean;
 
     /**
-     * Text color of selected Dock item.
+     * Used to specify a custom component to render as the dock button.
      */
-    selectedTextColor?: string;
+    dockButton?: DockButton;
 
     /**
-     * Background color of selected Dock item.
+     * Used to specify a custom component to render as a menu for the button.
      */
-    selectedBackgroundColor?: string;
+    dockMenu?: DockMenu;
+}
 
+export interface DockButton {
```

Comment:
The interface `DockButton` and `DockMenu` is exactly same. The other thing I would like to say here is we might not need `dockButton?: DockButton;` for now at least. It is better to add the property when we need it. 
Having and not using it confuses the consumer. What do you think?

Reply:
Yeah, true. I think removing the option to have a custom dock-button is better than keeping it and merging the two interfaces, because I think the latter would be even more confusing.

Reply:
I'll copy this to the thread I made about it, and hide this thread, so we can keep it in one place 👍